src/client.rs

```diff
@@ -963,84 +986,22 @@ mod tests {
 
     #[meilisearch_test]
     async fn test_update_key(client: Client, description: String) {
-        let expires_at = OffsetDateTime::now_utc() + time::Duration::HOUR;
-        let key = KeyBuilder::new(description.clone());
+        let mut key = KeyBuilder::new();
+        key.with_name("test_update_key");
         let mut key = client.create_key(key).await.unwrap();
 
-        key.actions = vec![Action::DocumentsAdd];
-        key.expires_at = Some(expires_at);
-        key.indexes = vec!["*".to_string()];
+        let name = "new name".to_string();
+        key.with_description(&description);
+        key.with_name(&name);
 
-        let key = client.update_key(key).await.unwrap();
+        let key = key.update(&client).await.unwrap();
 
-        assert_eq!(key.actions, vec![Action::DocumentsAdd]);
-        assert_eq!(key.description, description);
-        // We can't compare the two timestamp directly because of some nanoseconds imprecision with the floats
-        assert_eq!(
-            key.expires_at.unwrap().unix_timestamp(),
-            expires_at.unix_timestamp()
-        );
-        assert_eq!(key.indexes, vec!["*".to_string()]);
-
-        let keys = client.get_keys().await.unwrap();
-
-        let remote_key = keys.iter().find(|k| k.key == key.key).unwrap();
-
-        assert_eq!(remote_key.actions, vec![Action::DocumentsAdd]);
-        assert_eq!(remote_key.description, description);
-        // We can't compare the two timestamp directly because of some nanoseconds imprecision with the floats
-        assert_eq!(
-            remote_key.expires_at.unwrap().unix_timestamp(),
-            expires_at.unix_timestamp()
-        );
-        assert_eq!(remote_key.indexes, vec!["*".to_string()]);
+        assert_eq!(key.description, Some(description));
+        assert_eq!(key.name, Some(name));
```

Comment:
Same as before, I think we should use the string generation for the name and hardcode the description